lib/oidcstrategy.js

```diff
@@ -752,4 +574,468 @@ Strategy.prototype._shouldLoadUserProfile = function shouldLoadUserProfile(issue
   return done(null, !skip);
 };
 
+/**
+ * validate id_token, and pass the validated claims and the payload to callback
+ * if code (resp. access_token) is provided, we will validate the c_hash (resp at_hash) as well
+ *
+ * @param {String} id_token
+ * @param {Object} req
+ * @param {Function} callback
+ * @param {String} code (if you want to validate c_hash)
+ * @param {String} access_token (if you want o validate at_hash)
+ */
+Strategy.prototype._validateIdToken = function validateIdToken(id_token, code, access_token, req, callback) {
+  const self = this;
+
+  // decode id_token
+  const decoded = jws.decode(id_token);
+  if (decoded == null)
+    return self. fail(null, false, 'Invalid JWT token');
+  log.info('token decoded: ', decoded);
+
+  // get Pem Key
+  var PEMkey = undefined;
+  if (decoded.header.x5t) {
+    PEMkey = self.metadata.generateOidcPEM(decoded.header.x5t);
+  } else if (decoded.header.kid) {
+    PEMkey = self.metadata.generateOidcPEM(decoded.header.kid);
+  } else {
+    return self.fail('We did not reveive a token we know how to validate');
+  }
+
+  var options = self._options;
+
+  // since the id_token is for itself, so the id_token is the clientID by default
+  options.audience = options.clientID;
+
+  // verify id_token signature and claims
+  return jwt.verify(id_token, PEMkey, options, (err, jwtClaims) => {
```

Comment:
@lovemaths I don't see anything that guarantees that an issuer was set in the options.  Maybe I missed it.  I would prefer that we enforce that here, unless issuer validation has been explicitly turned off, so there is no mistake.